src/button.tsx

```diff
@@ -217,6 +219,62 @@ const useDangerButtonStyles = createUseStyles((theme) => ({
     inverse: dangerButtonStyles(theme),
 }));
 
+const renderButtonContent = ({
+    content,
+    defaultIconSize,
+    renderText,
+}: {
+    content: React.ReactNode;
+    defaultIconSize: number;
+    renderText: (text: React.ReactNode) => React.ReactNode;
+}): React.ReactNode => {
+    const length = React.Children.count(content);
+    const childrenArr = React.Children.toArray(content);
+    const resultChildrenArr: Array<React.ReactNode> = [];
+    let accText: Array<React.ReactNode> = [];
+    const flushAccText = () => {
+        resultChildrenArr.push(
+            <React.Fragment key={resultChildrenArr.length}>{renderText(accText)}</React.Fragment>
+        );
+        accText = [];
+    };
+    childrenArr.forEach((element, idx) => {
+        const isFirstChild = idx === 0;
+        const isLastChild = idx === length - 1;
+
+        const isIconElement =
+            React.isValidElement(element) &&
+            typeof element.type !== 'string' &&
+            element.type.name?.startsWith('Icon'); // all mistica icons start with Icon prefix
```

Comment:
You could be passing an element named IconWhatever that is not a mistica icon

Reply:
that's fine, that element would likely be an icon too